boot/boot.js

```diff
@@ -498,10 +503,32 @@ $tw.utils.getTypeEncoding = function(ext) {
 	return typeInfo ? typeInfo.encoding : "utf8";
 };
 
+var polyfill =[
+	"// this polyfills the globalThis variable",
+	"// using the this variable on a getter ",
+	"// inserted into the prototype of globalThis",
+	"(function() {",
+	"	if (typeof globalThis === 'object') return;",
+	"	// node.green says this is available since 0.10.48",
+	"	Object.prototype.__defineGetter__('__wow__', function() {",
+	"		return this;",
+	"	});",
+	"	__wow__.globalThis = __wow__; // lolwat",
+	"	delete Object.prototype.__wow__;",
+	"}());"
+].join("\n");
+
+var globalCheck =[
+	"  if(Object.keys(globalThis).length){",
+	"    console.log(Object.keys(globalThis));",
+	"    throw \"Global assignment is not allowed within modules on node.\";",
+	"  }"
+].join('\n');
```

Comment:
Am I right in understanding that this polyfill and globalcheck are needed as part of attempts to trap assignments to global variables? If so, perhaps it's better to drop them in the interests of simplicity.

Reply:
Yes, does it not matter if the global gets assignments?

Reply:
Yes, the idea here is to throw an error if anything tries to assign something to global so that we don't just silently ignore it and then make everyone wonder why it's not working.